pkg/scheduler/eventhandlers_test.go

```diff
@@ -524,3 +524,85 @@ func TestAdmissionCheck(t *testing.T) {
 		})
 	}
 }
+
+func TestNodeSchedulingPropertiesChange(t *testing.T) {
+	testCases := []struct {
+		name      string
+		newNode   *v1.Node
+		oldNode   *v1.Node
+		wantEvent *framework.ClusterEvent
+	}{
+		{
+			name:      "no specific changed applied",
+			newNode:   st.MakeNode().Unschedulable(false).Obj(),
+			oldNode:   st.MakeNode().Unschedulable(false).Obj(),
+			wantEvent: nil,
+		},
+		{
+			name:      "only node spec unavailable changed",
+			newNode:   st.MakeNode().Unschedulable(false).Obj(),
+			oldNode:   st.MakeNode().Unschedulable(true).Obj(),
+			wantEvent: &queue.NodeSpecUnschedulableChange,
+		},
+		{
+			name: "only node allocatable changed",
+			newNode: st.MakeNode().Capacity(map[v1.ResourceName]string{
+				v1.ResourceCPU:                     "1000m",
+				v1.ResourceMemory:                  "100m",
+				v1.ResourceName("example.com/foo"): "1"},
+			).Obj(),
+			oldNode: st.MakeNode().Capacity(map[v1.ResourceName]string{
+				v1.ResourceCPU:                     "1000m",
+				v1.ResourceMemory:                  "100m",
+				v1.ResourceName("example.com/foo"): "2"},
+			).Obj(),
+			wantEvent: &queue.NodeAllocatableChange,
+		},
+		{
+			name:      "only node label changed",
+			newNode:   st.MakeNode().Label("foo", "bar").Obj(),
+			oldNode:   st.MakeNode().Label("foo", "fuz").Obj(),
+			wantEvent: &queue.NodeLabelChange,
+		},
+		{
+			name: "only node taint changed",
+			newNode: st.MakeNode().Taints([]v1.Taint{
+				{Key: v1.TaintNodeUnschedulable, Value: "", Effect: v1.TaintEffectNoSchedule},
+			}).Obj(),
+			oldNode: st.MakeNode().Taints([]v1.Taint{
+				{Key: v1.TaintNodeUnschedulable, Value: "foo", Effect: v1.TaintEffectNoSchedule},
+			}).Obj(),
+			wantEvent: &queue.NodeTaintChange,
+		},
+		{
+			name:    "only node condition changed",
+			newNode: st.MakeNode().Obj(),
+			oldNode: st.MakeNode().Condition(
+				v1.NodeReady,
+				v1.ConditionTrue,
+				"Ready",
+				"Ready",
+			).Obj(),
+			wantEvent: &queue.NodeConditionChange,
+		},
+		{
+			name: "both node label and node taint changed",
+			newNode: st.MakeNode().
+				Label("foo", "bar").
+				Taints([]v1.Taint{
+					{Key: v1.TaintNodeUnschedulable, Value: "", Effect: v1.TaintEffectNoSchedule},
+				}).Obj(),
+			oldNode: st.MakeNode().Taints([]v1.Taint{
+				{Key: v1.TaintNodeUnschedulable, Value: "foo", Effect: v1.TaintEffectNoSchedule},
+			}).Obj(),
+			wantEvent: &framework.ClusterEvent{Resource: framework.Node, ActionType: framework.UpdateNodeLabel | framework.UpdateNodeTaint, Label: "NodeUpdate"},
```

Comment:
do we have a `framework.UpdateNode` as well?

Reply:
Yes, I added it with this PR. Update with the `event.Label`.